stone_age/game_board/building_tile.py

```diff
@@ -0,0 +1,83 @@
+
+    def __init__(self, cards: Iterable[Union[simple_building.SimpleBuilding,
+        arbitrary_building.ArbitraryBuilding,
+        variable_building.VariableBuilding]]):
+        self.cards: list[Union[simple_building.SimpleBuilding,
+        arbitrary_building.ArbitraryBuilding,
+        variable_building.VariableBuilding]] = list(cards)
+        self.figures: Optional[PlayerOrder] = None
+
+
+    def place_figures(self, player: Player, figure_count: int) -> bool:
+        if self.try_to_place_figures(player, figure_count) != HasAction.NO_ACTION_POSSIBLE:
+            player.player_board.take_figures(1)
+            self.figures = player.player_order
+            return True
+        return False
+
+
+    def try_to_place_figures(self, player: Player, count: int) -> HasAction:
+        assert isinstance(player, Player) and count > 0
+        if count > 1:
+            return HasAction.NO_ACTION_POSSIBLE
+        if not player.player_board.has_figures(count):
+            return HasAction.NO_ACTION_POSSIBLE
+        if self.figures is not None:
+            return HasAction.NO_ACTION_POSSIBLE
+        if not self.cards:
+            return HasAction.NO_ACTION_POSSIBLE
+        return HasAction.AUTOMATIC_ACTION_DONE
+
+    def make_action(self, player: Player, input_resources: Iterable[Effect],
+                    output_resources: Iterable[Effect]) -> ActionResult:
+        if self.figures is not None:
+            return ActionResult.FAILURE
+        if output_resources != input_resources:
+            return ActionResult.FAILURE
+        card: Union[simple_building.SimpleBuilding,
+        arbitrary_building.ArbitraryBuilding,
+        variable_building.VariableBuilding] = self.cards[-1]
+        result: Optional[int] = card.build(input_resources)
+        if (result is None or
+                self.try_to_make_action(player) == HasAction.NO_ACTION_POSSIBLE):
+            return ActionResult.FAILURE
+        if player.player_board.take_resourses(input_resources):
+            player.player_board.give_figure()
+#           player.player_board.give_card()
+#           player.player_board.give_points(result)
+            self.cards.pop()
+            return ActionResult.ACTION_DONE
+        return ActionResult.FAILURE
+
+    def skip_action(self, player: Player) -> bool:
+        if self.figures is not None and player.player_order != self.figures:
+            return False
+        if self.figures is None:
+            return True
+        player.player_board.give_figure()
+        self.figures = None
+        return True
+
+    def try_to_make_action(self, player: Player) -> HasAction:
+        assert isinstance(player, Player)
+        if player.player_order != self.figures:
+            return HasAction.NO_ACTION_POSSIBLE
+        if not self.cards:
+            return HasAction.NO_ACTION_POSSIBLE
+        if not self.figures:
+            return HasAction.NO_ACTION_POSSIBLE
+        return HasAction.AUTOMATIC_ACTION_DONE
```

Comment:
Here return `HasAction.WAITING_FOR_PLAYER_ACTION`, automatic action done would imply player already built the building, but this will only happen when he pays (we are waiting until he pays). Note that this method is used also in checking of game phases, if you return automatic action done, game state will think that building is built and may move to feeding tribe, for example (maybe this is not true as a whole, but it illustrates a bit of work of game phases).